src/Microsoft.Dnx.Compilation.CSharp/RoslynProjectReference.cs

```diff
@@ -68,45 +69,153 @@ public IList<ISourceReference> GetSources()
                                      .ToList();
         }
 
-        public Assembly Load(IAssemblyLoadContext loadContext)
+        private IEnumerable<ResourceDescriptor> GetResourcesForCulture(string cultureName)
         {
-            using (var pdbStream = new MemoryStream())
-            using (var assemblyStream = new MemoryStream())
-            {
-                IList<ResourceDescription> resources = CompilationContext.Resources;
+            var resourcesByCultureName = CompilationContext.Resources
+                .GroupBy(GetResourceCultureName, StringComparer.OrdinalIgnoreCase);
 
-                Logger.TraceInformation("[{0}]: Emitting assembly for {1}", GetType().Name, Name);
+            return resourcesByCultureName
+                .SingleOrDefault(grouping => string.Equals(grouping.Key, cultureName, StringComparison.OrdinalIgnoreCase));
+        }
 
-                var sw = Stopwatch.StartNew();
+        private string GetResourceCultureName(ResourceDescriptor res)
```

Comment:
This code needs to be heavily tested. It doesn't belong in the roslyn specific code. It's generic logic that should be tested in isolated and used by the roslyn compiler (or somehow merged into ResourceDescriptor)